src/maze/graphnode.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cstdint>
+
+constexpr uint8_t EAST = 0;
+constexpr uint8_t SOUTH = 1;
+constexpr uint8_t WEST = 2;
+constexpr uint8_t NORTH = 3;
+
+constexpr uint8_t DIRECTIONS = 4;
+constexpr uint8_t OPPOSITE = 2;
+
+/**
+ * Nodo del grafo.
+ */
+class GraphNode
```

Comment:
It is more compact to create the class with `struct` instead, move the private section to the bottom, and omit the `public` keyword.

Putting public before private allows driving the focus to the public interface of a class, as it should be, and it is an approach suggested for example in Robert C. Martin's "Clean Code" book.